pkg/controller/projectclaim/customresourceadapter.go

```diff
@@ -137,13 +138,13 @@ func (c *CustomResourceAdapter) EnsureProjectClaimState(state gcpv1alpha1.ClaimS
 
 	if state == gcpv1alpha1.ClaimStatusPending {
 		if c.projectClaim.Status.State != "" {
-			return nil
+			return fmt.Errorf("Unexpected ProjectClaim state transition: %s -> %s", c.projectClaim.Status.State, state)
```

Comment:
As discussed, there may be occasions where state is `ClaimStatusPendingProject`, which we need to return `nil` for.